example/m5_competition.py

```diff
@@ -0,0 +1,304 @@
+calendar_raw["snap_TX"] = calendar_raw["snap_TX"].astype(np.int32)
+calendar_raw["snap_WI"] = calendar_raw["snap_WI"].astype(np.int32)
+
+# TODO: Make the case in Temporian. Float32 is required for the SMA operator.
+sales_raw["sales"] = sales_raw["sales"].astype(np.float32)
+
+# Convert to Temporian Events
+#
+# We can finally convert the dataset to Temporian format.
+
+print("Convert to Temporian Events")
+print("===========================")
+
+# TODO: Remove
+print("sales_raw:")
+sales_raw.info()
+print("sell_prices_raw:")
+sell_prices_raw.info()
+print("calendar_raw:")
+calendar_raw.info()
+
+# TODO: Do the grouping by index during the processing.
+sales_data = tp.EventData.from_dataframe(
+    sales_raw,
+    index_names=["item_id", "dept_id", "cat_id", "store_id", "state_id"],
+)
+
+calendar_data = tp.EventData.from_dataframe(calendar_raw)
+
+sell_prices_data = tp.EventData.from_dataframe(
+    sell_prices_raw,
+    index_names=["store_id", "item_id"],
+)
+
+print("sales_data:\n", sales_data)
+print("sell_prices_data:\n", sell_prices_data)
+print("calendar_data:\n", calendar_data)
+
+# How that out data is in the Temporian format. We can delete the Pandas
+# dataframe objects.
+
+print("Release Pandas dataframes")
+print("===========================")
+
+del sales_raw
+del sell_prices_raw
+del calendar_raw
+
+# Plot raw data
+print("Plot raw data")
+print("============")
+
+plot_options = {
+    # Only plot the first index / first product.
+    "index": None,
+    # We only plot 1year of data to make the plot more readable.
+    "min_time": datetime(2015, 1, 1),
+    "max_time": datetime(2016, 1, 1),
+}
+
+sales_data.plot(**plot_options).savefig(
+    os.path.join(work_directory, "raw_sales.png")
+)
+calendar_data.plot(**plot_options).savefig(
+    os.path.join(work_directory, "raw_calendar.png")
+)
+sell_prices_data.plot(**plot_options).savefig(
+    os.path.join(work_directory, "raw_sell_prices.png")
+)
+
+# Convert time series to tabular dataset
+print("Convert time series to tabular dataset")
+print("======================================")
+
+# We define the computation graph
+sales = sales_data.schema()
+calendar = calendar_data.schema()
+sell_prices = sell_prices_data.schema()
+
+augmented_sales = tp.glue(
+    # Raw sales
+    sales,
+    # Moving average of sales
+    tp.prefix("sma_7.", tp.simple_moving_average(sales, tp.duration.days(7))),
+    tp.prefix("sma_28.", tp.simple_moving_average(sales, tp.duration.days(28))),
+    tp.prefix("sma_84.", tp.simple_moving_average(sales, tp.duration.days(84))),
+    # Sum of sales
+    tp.prefix("sum_7.", tp.moving_sum(sales, tp.duration.days(7))),
+    tp.prefix("sum_28.", tp.moving_sum(sales, tp.duration.days(28))),
+    tp.prefix("sum_84.", tp.moving_sum(sales, tp.duration.days(84))),
+)
+
+# Lagged sales
+lagged_sales = []
+for lag in [1, 2, 3]:
+    lagged_sales.append(
+        tp.sample(
+            tp.prefix(f"lag_{lag}.", tp.lag(sales, tp.duration.days(lag))),

```

Comment:
Could we take advantage of the fact that `tp.lag()` can take as input an array of events?

Reply:
From chat., I propose to keep it this way for now.